protocol/kafka_broker/broker.go

```diff
@@ -105,10 +134,38 @@ func (b *Broker) Close() error {
 	return nil
 }
 
-func (b *Broker) SendAndReceive(request []byte) (Response, error) {
+func (b *Broker) SendAndReceive(request builder.RequestI, stageLogger *logger.Logger) (Response, error) {
+	var apiType string
+	var apiVersion int16
+	var correlationId int32
+	var message []byte
+
+	switch req := request.(type) {
+	case *kafkaapi.ApiVersionsRequest:
```

Comment:
We aren't using interfaces properly here - explained this over call, let's reword